test/test_e2e.py

```diff
@@ -0,0 +1,337 @@
+            time_frames=[ETimeFrames.MID],
+            scopes=[EScope.S1S2, EScope.S1S2S3],
+            aggregation_method=PortfolioAggregationMethod.WATS,
+        )
+
+        portfolio_data = SBTi.utils.get_data([data_provider], pf_companies)
+        scores = temp_score.calculate(portfolio_data)
+        agg_scores = temp_score.aggregate_scores(scores)
+
+        # verify that results exist
+        self.assertAlmostEqual(agg_scores.mid.S1S2.all.score, 0.61666, places=4)
+
+    def test_basic_flow(self):
+        """
+        This test is going all the way to the aggregated calculations
+        """
+
+        companies, targets, pf_companies = self.create_base_companies(["A", "B"])
+
+        data_provider = TestDataProvider(companies=companies, targets=targets)
+
+        # Calculate scores & Aggregated values
+        temp_score = TemperatureScore(
+            time_frames=[ETimeFrames.MID],
+            scopes=[EScope.S1S2, EScope.S1S2S3],
+            aggregation_method=PortfolioAggregationMethod.WATS,
+        )
+
+        portfolio_data = SBTi.utils.get_data([data_provider], pf_companies)
+        scores = temp_score.calculate(portfolio_data)
+        agg_scores = temp_score.aggregate_scores(scores)
+
+        # verify that results exist
+        self.assertEqual(agg_scores.mid.S1S2.all.score, 0.43)
+
+    # Run some regression tests
+    # @unittest.skip("only run for longer test runs")
+    def test_regression_companies(self):
+
+        nr_companies = 1000
+
+        # test 10000 companies
+        companies: List[IDataProviderCompany] = []
+        targets: List[IDataProviderTarget] = []
+        pf_companies: List[PortfolioCompany] = []
+
+        for i in range(nr_companies):
+            company_id = f"Company {str(i)}"
+            # company
+            company = copy.deepcopy(self.company_base)
+            company.company_id = company_id
+            companies.append(company)
+
+            # target
+            target = copy.deepcopy(self.target_base)
+            target.company_id = company_id
+            targets.append(target)
+
+            # pf company
+            pf_company = PortfolioCompany(
+                company_name=company_id,
+                company_id=company_id,
+                investment_value=100,
+                company_isin=company_id,
+            )
+            pf_companies.append(pf_company)
+
+        data_provider = TestDataProvider(companies=companies, targets=targets)
+
+        # Calculate scores & Aggregated values
+        temp_score = TemperatureScore(
+            time_frames=[ETimeFrames.MID],
+            scopes=[EScope.S1S2],
+            aggregation_method=PortfolioAggregationMethod.WATS,
+        )
+
+        portfolio_data = SBTi.utils.get_data([data_provider], pf_companies)
+        scores = temp_score.calculate(portfolio_data)
+        agg_scores = temp_score.aggregate_scores(scores)
+
+        self.assertAlmostEqual(agg_scores.mid.S1S2.all.score, 0.43)
```

Comment:
I would put the `0.43` score somewhere as a variable in the constructor (or so) that you don't need to hardcode it at many places and is easy to change in case something will change reg. the companies/targets you use